### PR TITLE
Rattahamberg/courses layout

### LINKED ISSUE #8: Courses Layout
image

Donl't worry about the courses, that's another issue. Only work on what's in the screenshot

### PR DESCRIPTION
**Pull request by rattahamberg**
image

### CODE REVIEW

**Review by rattahamberg — COMMENTED** [review]
Formatting touched all of these files for some reason; I didn't change any of this manually.

File: tailwind.config.ts [hunk 1]
```diff
@@ -2,7 +2,14 @@
 module.exports = {
   content: ["./src/**/*.{ts,tsx}"],
   theme: {
-    extend: {},
+    extend: {
+      colors: {
+        "course-layout-background": "#498A39",
+      },
```
Comment (stanleylew5) [inline tailwind.config.ts]: solve your merge conflict, if you don't know how we'll cover this tomorrow

File: src/components/courselayout.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+interface CourseLayoutProps {
+  title: string;
+  description: string;
+}
+
```
Comment (stanleylew5) [inline src/components/courselayout.tsx]: props do not need to be passed here, sorry for having unclear issues but we'll replace the lorem ipsum/description with actual text soon but you can just put lorem ipsum text for now

File: src/components/courselayout.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const CourseLayout: React.FC<CourseLayoutProps> = ({ title, description }) => {
+  return (
+    <div className="mx-auto flex w-full max-w-[calc(100%-100px)] flex-col items-center rounded-[20px] bg-course-layout-background/80 p-8 text-center font-Malayalam font-normal">
```
Comment (stanleylew5) [inline src/components/courselayout.tsx]: the tailwind classes that are being used here are overcomplicated please ask for help tomorrow

File: src/components/courselayout.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      <div className="pb-4 pt-8 text-center font-Malayalam text-[80px] font-normal leading-[100px]">
```
Comment (stanleylew5) [inline src/components/courselayout.tsx]: values here are hardcoded like 80 px, use tailwind classes instead

File: src/app/page.tsx [hunk 5]
```diff
@@ -1,7 +1,12 @@
+import CourseLayout from "@/components/courselayout";
+
 const Home = () => {
   return (
     <div className="flex h-screen w-screen items-center justify-center bg-red-500 text-white">
-      Hello World
+      <CourseLayout
+        title="Our Projects"
```
Comment (stanleylew5) [inline src/app/page.tsx]: no props need to be passed

**Review by stanleylew5 — CHANGES_REQUESTED** [review]
delete entire .idea folder.

**Comment by rattahamberg** [comment]
Fixed (I think) all requested changes.

File: src/components/courselayout.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const CourseLayout = () => {
+  return (
+    <div className="bg-transparent-80 p-auto font-Malayalam mx-auto flex w-[95vw] flex-col items-center rounded-[1vw] bg-devstation-green-200 p-[2.5vw] text-center font-normal">
+      <div className="text-[4vw]">Our Projects</div>
+
+      <div className="text-[1.15vw]">
```
Comment (stanleylew5) [inline src/components/courselayout.tsx]:
```suggestion
      <div className="text-xl">
```
this isn't technically "header" text we can just leave it as text-xl

File: src/components/courselayout.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const CourseLayout = () => {
+  return (
+    <div className="bg-transparent-80 p-auto font-Malayalam mx-auto flex w-[95vw] flex-col items-center rounded-[1vw] bg-devstation-green-200 p-[2.5vw] text-center font-normal">
```
Comment (stanleylew5) [inline src/components/courselayout.tsx]:
```suggestion
    <div className="bg-transparent-80 p-auto mx-auto flex w-[95vw] flex-col items-center rounded-xl bg-devstation-green-200 p-[2.5vw]">
```

Comment (stanleylew5) [inline src/components/courselayout.tsx]:
```suggestion
    <div className="bg-transparent-80 w-11/12 flex flex-col items-center rounded-xl bg-devstation-green-200 p-[2.5vw]">
```
a lot of the classes you added are not needed

File: src/components/courselayout.tsx (hunk 6 above)
Comment (stanleylew5) [inline src/components/courselayout.tsx]:
```suggestion
      <div className="text-lg">
```
this is technically not header text, just the above was

File: src/components/courselayout.tsx (hunk 7 above)
Comment (stanleylew5) [reply src/components/courselayout.tsx]: if you're curious as to why, feel free to ask us in the general chat

**Review by stanleylew5 — CHANGES_REQUESTED** [review]

**Comment by rattahamberg** [comment]
Simplified the CSS some more.

File: src/components/courselayout.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const CourseLayout = () => {
+  return (
+    <div className="bg-transparent-80 flex w-11/12 flex-col items-center rounded-xl bg-devstation-green-200 p-[2.5vw] text-center">
+      <div className="text-[4vw]">Our Projects</div>
+
+      <div className="text-[1.15vw]">
```
Comment (stanleylew5) [inline src/components/courselayout.tsx]:
```suggestion
      <div className="text-lg">
```

**Review by stanleylew5 — CHANGES_REQUESTED** [review]

**Pull request merged**